### ISSUE TITLE
Does it support batch norm layer in current version?

### ISSUE THREAD

**Issue by ctchu** [BVLC/caffe#5456]
Does current version have batch norm layer? When I check src/caffe/prot/caffe.proto, there is no BatchNorm defined. However, on the webpage of caffe (http://caffe.berkeleyvision.org/tutorial/layers/batchnorm.html), it says it is available. When I add BatchNorm  into my prototxt, it gave me the error below, so I guess batch norm is not supported here?

[libprotobuf ERROR C:\Users\guillaume\work\build_v140_x64\packages\protobuf\protobuf_download-prefix\src\protobuf_download\src\google\protobuf\text_format.cc:298] Error parsing text-format caffe.NetParameter: 63:9: Unknown enumeration value of "BATCHNORM" for field "type".

Thanks,

Randy

**Comment by ShaggO** [comment]
Caffe changed to using strings instead of enumerates quite a while back. Try using "BatchNorm" in the prototxt instead.

**Comment by williford** [comment]
caffe.proto does mention BatchNorm. See:
https://github.com/BVLC/caffe/blob/master/src/caffe/proto/caffe.proto

As @ShaggO stated, the type is "BatchNorm" (see the "Layer type" field in the tutorial page you referenced).

You can also see: http://caffe.help/manual/layers/batchnorm.html
